ClientApp/api/axiosInstance.ts

```diff
@@ -114,47 +89,64 @@ axiosInstance.interceptors.request.use(
   (error) => Promise.reject(error)
 );
 
-// Response Interceptor for Error Handling
-// ****************************************
 axiosInstance.interceptors.response.use(
   (response: AxiosResponse) => response,
   async (error) => {
     const originalRequest = error.config;
 
-    // Network error or server down
     if (!error.response) {
-      showAlert(ALERT_MESSAGES.networkError.title, ALERT_MESSAGES.networkError.message);
-      return Promise.reject(error);
-    }
-
-    // Handle 401 Unauthorized for token refresh failure
-    if (error.response.status === 401 && originalRequest.url === `${config.baseURL}${API_ENDPOINTS.REFRESH_TOKEN}`) {
-      return Promise.reject(error);
-    }
-
-    // Handle 401 Unauthorized due to expired or invalid token
-    if (error.response.status === 401 && error.response.data?.code === 'token_not_valid') { 
-      const newAccessToken = await refreshAccessToken();
-      if (newAccessToken) {
-        originalRequest.headers.Authorization = `Bearer ${newAccessToken}`;
-        return axiosInstance(originalRequest);  // Retry the original request with new token
-      } else {
-        showAlert(ALERT_MESSAGES.sessionExpired.title, ALERT_MESSAGES.sessionExpired.message);
+      // Network error or no response
+      consoleError(ALERT_MESSAGES.networkError.title, ALERT_MESSAGES.networkError.message);
+    } else {
+      const { status } = error.response;
+      
+      // Specific handling for common HTTP errors, including status code in the message
+      switch (status) {
+        case 400:
+          consoleError(ALERT_MESSAGES.badRequest.title, ALERT_MESSAGES.badRequest.message, status);
+          break;
+        case 401:
+          const newAccessToken = await refreshAccessToken();
+          if (newAccessToken) {
+            originalRequest.headers.Authorization = `Bearer ${newAccessToken}`;
+            return axiosInstance(originalRequest); // Retry original request
+          } else {
+            consoleError(ALERT_MESSAGES.sessionExpired.title, ALERT_MESSAGES.sessionExpired.message, status);
+            
+            // Logout user
```

Comment:
We can encapsulate the logout logic here, maybe me centralize it somewhere else. It could be something like:
```
const handleLogout = async () => {
  try {
    await AsyncStorage.multiRemove(['accessToken', 'refreshToken']);
  } catch (error) {
    console.error('Error during logout:', error);
  }
};
```

If I am not mistaken I think you did a logout method somewhere, you can maybe use that?

Reply:
Good note! this will be addressed soon by implementing global events

I have to think about how to do that, because the function `logout` that I have defined in `AuthContext` can't be used in the `axiosInstance` because `AuthContext` imports `axiosInstance` and If I use a function from `AuthContext` in `axiosInstance` it will lead to circular imports which will cause faulty behaviour..I need to centralize the logic somewhere else but for now this is ok for user login and logout